aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -507,7 +507,7 @@ Tensor to_dense_backward(const Tensor& grad, const Tensor& input_) {
     // TODO: implement sparse_compressed_mask
     switch(input_.layout()) {
     case kSparseCsr: return grad.sparse_mask(input_.to_sparse()).to_sparse_csr();
-    case kSparseCsc: return grad.sparse_mask(input_.to_sparse()).to_sparse_csc();
+    case kSparseCsc: return grad.sparse_mask(input_.to_sparse().coalesce()).to_sparse_csc();
```

Comment:
In general, 
- `sparse_mask` should be able to handle the uncoalesced input,
- `sparse_mask` should be able to handle sparse compressed tensors as input.

However, both features are too big for this bug-fix PR to be handled, so consider this note OT.